dsp/transform/hilbert.go

```diff
@@ -0,0 +1,89 @@
+// Copyright ©2024 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package transform
+
+import (
+	"math/cmplx"
+
+	"gonum.org/v1/gonum/dsp/fourier"
+)
+
+// Implements an approximate Hilbert transform that allows the user to determine an
+// approximate analytical signal of a real signal, and determine the real 'envelope'
+// of a signal.
+//
+// The underlying implementation uses a DFT and IDFT transform, so usual methods for
+// speeding up DFT/IDFT are likely to apply here as well.
```

Comment:
```suggestion
// Hilbert implements an approximate Hilbert transform that allows calculation
// of an approximate analytical signal of a real signal, and determine the
// real envelope of a signal.
//
// The underlying implementation uses a discrete Fourier transform and inverse
// and discrete Fourier transform, so usual methods for speeding up these
// transforms are likely to apply here as well.
```

Is the second paragraph directed at users? If so, a link to a relevant article would be nice. If they are directed at us as maintainers, the comment should not be visible in the godoc.

Reply:
Made this comment nicer, and removed the second paragraph. I added another comment to the `NewHilbert` method noting a performance tip. My source for that performance tip is a comment [in the internal/fftpack source](https://github.com/gonum/gonum/blob/1ca563a018b641e805317f1ac9ae0d37b32d162c/dsp/fourier/internal/fftpack/cfft.go#L126)